logzio-sender/src/main/java/io/logz/sender/LogzioSender.java

```diff
@@ -19,85 +17,101 @@
-                .builder()
-                .setLogzioToken(logzioToken)
-                .setLogzioType(logzioType)
-                .setLogzioListenerUrl(logzioUrl)
-                .setSocketTimeout(socketTimeout)
-                .setConnectTimeout(connectTimeout)
-                .setCompressRequests(compressRequests)
-                .build();
+        if (logsBuffer == null || reporter == null || httpsRequestConfiguration == null) {
+            throw new LogzioParameterErrorException("logsBuffer=" + logsBuffer + " reporter=" + reporter
+                    + " httpsRequestConfiguration=" + httpsRequestConfiguration ,
+                    "For some reason could not initialize URL. Cant recover..");
+        }
 
+        this.logsBuffer = logsBuffer;
         this.drainTimeout = drainTimeout;
-        this.fsPercentThreshold = fsPercentThreshold;
         this.debug = debug;
-        this.gcPersistedQueueFilesIntervalSeconds = gcPersistedQueueFilesIntervalSeconds;
         this.reporter = reporter;
         httpsSyncSender = new HttpsSyncSender(httpsRequestConfiguration, reporter);
-        if (this.fsPercentThreshold == -1) {
-            dontCheckEnoughDiskSpace = true;
-        }
-        // divide bufferDir to dir and queue name
-        if (bufferDir == null) {
-            throw new LogzioParameterErrorException("bufferDir", "value is null.");
-        }
-        String dir = bufferDir.getAbsoluteFile().getParent();
-        String queueNameDir = bufferDir.getName();
-        if (dir == null || queueNameDir.isEmpty() ) {
-            throw new LogzioParameterErrorException("bufferDir", " value is empty: "+bufferDir.getAbsolutePath());
-        }
-        logsBuffer = new BigQueue(dir, queueNameDir);
-        queueDirectory = bufferDir;
         this.tasksExecutor = tasksExecutor;
-
         debug("Created new LogzioSender class");
     }
 
+    @Deprecated
+    public static synchronized LogzioSender getOrCreateSenderByType(String logzioToken, String logzioType,
+                                                                    int drainTimeout, int fsPercentThreshold,
+                                                                    File bufferDir, String logzioUrl, int socketTimeout,
+                                                                    int connectTimeout, boolean debug,
+                                                                    SenderStatusReporter reporter,
+                                                                    ScheduledExecutorService tasksExecutor,
+                                                                    int gcPersistedQueueFilesIntervalSeconds,
+                                                                    boolean compressRequests)
+            throws LogzioParameterErrorException {
+
+
+        LogzioLogsBufferInterface logsBuffer = null;
+        if (bufferDir != null) {
+            logsBuffer = DiskQueue
+                    .builder(null, null)
```

Comment:
This line doesn't make sense due to null, null

Reply:
@asafm it's just for backward compatibility